Feature/handled exceptions properties including ios

TroubleMakerBen opened the pull request:
Added bindings for `trackModelException:withProperties:`.

TroubleMakerBen commented on SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Crashes.cs:
```diff
@@ -174,6 +173,11 @@ static string AnonymizePath(string path)
+        {
+            return NSDictionary.FromObjectsAndKeys(dict.Values.ToArray(), dict.Keys.ToArray());
+        }
+
```
Let me know if you prefer this somewhere else.

guperrot replied:
Can you check what we do on Analytics.TrackEvent? its a 1 liner so I am not too worried about duplication just make sure it's managed the exact same way.

TroubleMakerBen commented:
@guperrot Addressed your comment about making the logic identical to Analytics.

Yes, I tested if this works.

No Properties:
```
  "sid" : "1CAC92CE-8696-4EAE-8182-81646E67463C",
  "timestamp" : "2018-01-23T19:42:55.793Z",
  "device" : {
    "appVersion" : "1.1.1",
    "appBuild" : "1.1.1",
    "osName" : "iOS",
    "timeZoneOffset" : -480,
    "wrapperSdkVersion" : "1.1.1-SNAPSHOT",
    "osVersion" : "11.1",
    "wrapperRuntimeVersion" : "11.6.1",
    "locale" : "en_US",
    "wrapperSdkName" : "appcenter.xamarin",
    "appNamespace" : "com.microsoft.appcenter.xamarin.forms.puppet",
    "osBuild" : "16G1114",
    "sdkName" : "appcenter.ios",
    "oemName" : "Apple",
    "sdkVersion" : "1.2.1-7",
    "model" : "x86_64",
    "screenSize" : "1334x750"
  },
  "type" : "handledError",
  "id" : "04467DE7-845A-4632-AE39-C46AC74E2CC4",
  "exception" : {
    "message" : "Attempted to divide by zero.",
    "wrapperSdkName" : "appcenter.xamarin",
    "type" : "System.DivideByZeroException",
    "stackTrace" : "  at Contoso.Forms.Puppet.CrashesContentPage+<>c.<DivideByZero>b__13_0 () [0x00000] in /Users/benny/Repositories/MS/mobile-center-sdk-dotnet/Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs:137 \n  at Contoso.Forms.Puppet.CrashesContentPage.HandleOrThrow (System.Action action) [0x00002] in /Users/benny/Repositories/MS/mobile-center-sdk-dotnet/Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs:120 "
  }
```

With Properties:
```
{
  "device" : {
    "appVersion" : "1.1.1",
    "appBuild" : "1.1.1",
    "osName" : "iOS",
    "timeZoneOffset" : -480,
    "wrapperSdkVersion" : "1.1.1-SNAPSHOT",
    "osVersion" : "11.1",
    "wrapperRuntimeVersion" : "11.6.1",
    "locale" : "en_US",
    "wrapperSdkName" : "appcenter.xamarin",
    "appNamespace" : "com.microsoft.appcenter.xamarin.forms.puppet",
    "osBuild" : "16G1114",
    "sdkName" : "appcenter.ios",
    "oemName" : "Apple",
    "sdkVersion" : "1.2.1-7",
    "model" : "x86_64",
    "screenSize" : "1334x750"
  },
  "id" : "6811E817-CFC8-4E0B-A7AB-412A8F61DEB3",
  "sid" : "1CAC92CE-8696-4EAE-8182-81646E67463C",
  "exception" : {
    "message" : "Test crash exception generated by SDK",
    "wrapperSdkName" : "appcenter.xamarin",
    "type" : "Microsoft.AppCenter.Crashes.TestCrashException",
    "stackTrace" : "  at Microsoft.AppCenter.Crashes.Crashes.GenerateTestCrash () [0x00001] in /Users/benny/Repositories/MS/mobile-center-sdk-dotnet/SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Shared/Crashes.cs:158 \n  at Contoso.Forms.Puppet.CrashesContentPage+<>c.<TestException>b__12_0 () [0x00000] in /Users/benny/Repositories/MS/mobile-center-sdk-dotnet/Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs:130 \n  at Contoso.Forms.Puppet.CrashesContentPage.HandleOrThrow (System.Action action) [0x00002] in /Users/benny/Repositories/MS/mobile-center-sdk-dotnet/Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs:120 "
  },
  "timestamp" : "2018-01-23T19:43:23.549Z",
  "type" : "handledError",
  "properties" : {
    "Key1" : "Value1"
  }
}
```
